DataScience/Analytics/data/investigateCountyFiles.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python
+"""
+ONS Address Index - Investigating Different County Lists
+========================================================
+
+A simple script to compare two county lists i.e. to find the union, intersect, and complements.
+
+
+Running
+-------
+
+After all requirements are satisfied and the datasets are available, the script can be run using CPython::
+
+    python investigateCountyFiles.py
+
+
+Requirements
+------------
+
+:requires: pandas (tested with 0.19.2)
+
+
+Author
+------
+
+:author: Sami Niemi (sami.niemi@valtech.co.uk)
+
+
+Version
+-------
+
+:version: 0.1
+:date: 16-Mar-2017
+"""
+import pandas as pd
+
+
+def _normalize(df):
+    """
+    Normalise the county data i.e. strip spaces and convert to upper case, and finally sort alphabetically.
+
+    :param df: input data frame with county column
+    :type df: pandas.DataFrame
+
+    :return: normalised data
+    :rtype: pandas.DataFrame
+    """
+    df.sort_values(by='county', inplace=True)
+    df['county'] = df['county'].str.upper()
+    df['county'] = df['county'].str.strip()
+
+    return df
+
+
+def read_original_county_file():
+    """
+    Read in original county file that is present in the repository.
+    Normalises the county information.
+
+    :return: counties in a dataframe
+    :rtype: pandas.DataFrame
+    """
+    df = pd.read_csv('../../data/counties.csv')
+    df = _normalize(df)
+
+    print(len(df.index), 'counties in the original file')
+
+    return df
+
+
+def read_paul_county_file():
+    """
+    Read in Paul's county file. Normalises the county information.
+
+    :return: counties in a dataframe
+    :rtype: pandas.DataFrame
+    """
+    df = pd.read_csv('/Users/saminiemi/Dropbox/ONS/County_List_PG.csv')
```

Comment:
Not really, but the file is in Confluence not in the repo. I can change the filename to be a keyword arg so at least it's obvious that it requires changing...